File: backend/app/models/stock_movement.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Integer, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from ..database import Base


class StockMovement(Base):
    __tablename__ = "stock_movements"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    product_id = Column(String, ForeignKey("inventory.id"), nullable=False, index=True)
    quantity = Column(Integer, nullable=False)
    type = Column(String(10), nullable=False)
    reason = Column(Text, default="")
    before_stock = Column(Integer, nullable=False)
    after_stock = Column(Integer, nullable=False)
    user_id = Column(String, ForeignKey("users.id"), nullable=True)
    date = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    product = relationship("InventoryItem", back_populates="stock_movements")
    user = relationship("User", back_populates="stock_movements")

File: dotnet-backend/src/Javaline.Commercial.Domain/Entities/StockMovement.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Javaline.Commercial.Domain.Common;

namespace Javaline.Commercial.Domain.Entities;

[Table("stock_movements")]
public class StockMovement : BaseEntity
{
    [Column("product_id")]
    public string ProductId { get; set; } = string.Empty;

    [Column("quantity")]
    public decimal Quantity { get; set; }

    [Column("type")]
    public string Type { get; set; } = string.Empty;

    [Column("reason")]
    public string? Reason { get; set; }

    [Column("before_stock")]
    public decimal BeforeStock { get; set; }

    [Column("after_stock")]
    public decimal AfterStock { get; set; }

    [Column("user_id")]
    public string? UserId { get; set; }

    [Column("date")]
    public DateTime Date { get; set; } = DateTime.UtcNow;

    [JsonIgnore]
    public InventoryItem Product { get; set; } = null!;

    [JsonIgnore]
    public User? User { get; set; }
}
